### ISSUE TITLE
[bug] fluid-tailwind not working starting from v6.21.0

### ISSUE THREAD

**Issue by RafaelKr** [roots/bud#2639]
### Agreement

- [X] This is not a duplicate of an existing issue
- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Describe the issue

When using Tailwind with [fluid-tailwind](https://fluid.tw/) the fluid classes are not included in the compiled CSS. This issue started with bud 6.21.0, the fluid classes are available when using bud 6.20.0

It must be related to bud (or its dependencies). When using `npx bud tailwindcss` the classes are also correctly added to `public/styles/app.css`, even when using bud 6.21.0+

I created a reproduction repo at https://github.com/RafaelKr/sage-fluid-tailwind
The `main` branch contains the working state with bud 6.20.0, the `bud-6.21.0` branch contains the updated version (via `npx bud upgrade 6.21.0`) which doesn't include the fluid class.

I also ran `npx bud upgrade` and created a `bud-6.23.2` branch which also doesn't contain the fluid class.

### Expected Behavior

The compiled CSS should include the class selector `.\~p-6\/10` (escaped version of `~p-6/10`).

### Actual Behavior

The class is not in the compiled CSS.

### Steps To Reproduce

1. Run
	```sh
	git clone git@github.com:RafaelKr/sage-fluid-tailwind.git
	cd sage-fluid-tailwind
	yarn
	yarn build
	```
1. Check that `public/css/app.35925b.css` contains a selector containing `~p` 
	```
    cat public/css/app.35925b.css | grep '~p'
    ```
1. Run
	```sh
	git switch bud-6.23.2 # or bud-6.21.0
	yarn
	yarn build
	```
1. Now `public/css/app.53672d.css` (notice the changed hash) doesn't include the class selector containing `~p`. (If you used bud-6.21.0 the hash is `public/css/app.bbe229.css`)
	```
    cat public/css/app.53672d.css | grep '~p'
    # no output here
    ```

You can also run `npx bud tailwindcss` which generates `public/styles/app.css` containing the class, independently of the bud version

### version

6.23.2

**Comment by kellymears** [comment]
> The compiled CSS should include the class selector .\~p-6\/10 (escaped version of ~p-6/10).

Super helpful. I haven't reproduced this yet but this is a great issue.

**Comment by RafaelKr** [comment]
Thanks! I already tried to track it down myself, but after digging deeper for 4+ hours without results I gave up 😆

What do you mean with you haven't reproduced? Did you try it and it works for you also with the latest version?
Maybe also a `rm '~/.cache/bud-nodejs/*'` is required between rebuilds (shouldn't `--force` do the same? IIRC I tried that while debugging but didn't get the desired output, only after deleting that cache files with the command).

For debugging this stuff it would be super helpful to have bud create a version of the output files between every step of the pipeline, I tried if I can do this somehow, but quickly noticed it would take me much more time than I have to dedicate with my current knowledge of the bud codebase (and API).
Edit: Something like https://github.com/andywer/postcss-debug (although the project is extremely outdated and probably not compatible anymore)

**Comment by kellymears** [comment]
I just hadn't gotten around to actually reproducing it. I have now. 

Honestly, I think this is an upstream bug with either the plugin or tailwindcss. 

Tailwind supports passing a tailwind config object to postcss directly. If you do that with `fluid-tailwind` enabled the classes are stripped out even without using bud.js.

Here is a codesandbox reproducing the issue. This isn't a bud.js project, it's just a simple project using a standard `postcss.config.js` file: https://codesandbox.io/p/devbox/nice-wing-vvd8gf?file=%2Fpostcss.config.js%3A10%2C17. Obviously, postcss/tailwind are both working since the core tailwind classes render, but the fluid tailwindcss is stripped out in the same manner as your reproduction. 

However, if I configure tailwindcss using a path to the `tailwind.config.js` module then everything works. So, in [6938f59](https://github.com/roots/bud/commit/6938f598654d3d73850b2afcc8ad24e13e913f3e) I'm changing things so that we just pass the path as a string if the config is not modified using the `bud.tailwind` API.

This will fix your problem, but just to reiterate: the root cause is something upstream of us.

**Comment by RafaelKr** [comment]
Thanks! Could you maybe release a `6.23.3` with the fix? Because we're currently using a workaround where we first compile tailwind-fluid only and then import this to our main.css which requires to always manually recompile when new classes are added.

Edit: Do you have an idea if it is more likely an upstream bug of tailwind or the plugin? I'd like to "pass" the issue upstream. If you don't know I'd first try to create an issue at tailwind.

**Comment by RafaelKr** [comment]
Thank you @kellymears! I can confirm the fix works for me.
